### PR TITLE
Cleanup the SSO prototype code for merging

### PR DESCRIPTION
**Pull request by Dantemss**
Also added a separate secret_token setting for the SSO cookie and simplified the list of trusted hosts.

Merge at the same time as https://github.com/openstax/tutor-deployment/pull/435

### CODE REVIEW

**Comment by jpslav** [comment]
@Dantemss after this PR is ready, how will we reestablish a branch that still has the capistrano and other modifications so we can continue deploying to RDLS for experimentation?

File: lib/user_session_management.rb [hunk 1]
```diff
@@ -35,6 +44,11 @@ def sign_in!(user, options={})
 
   def sign_out!(options={})
     clear_pre_auth_state
+
+    # TODO move this to shared lib if this goes into production
+    session_config = Rails.application.secrets.sso['cookie']
+    cookies.delete(session_config['name'], domain: session_config['domain'])
```
[the remark this answers is not in the document]
Comment (Dantemss) [reply lib/user_session_management.rb]: Don't need to do this, I'll remove this comment

**Comment by Dantemss** [comment]
Re:
```
@Dantemss after this PR is ready, how will we reestablish a branch that still has the capistrano and other modifications so we can continue deploying to RDLS for experimentation?
```

I was imagining that Nathan would just continue adding more stuff to the other branch.

File: config/secrets.yml.example [hunk 2]
```diff
@@ -63,6 +63,7 @@ development:
     - \Acnx.org\z
     - localhost
     - 127.0.0.1
+    - .*\.prod\.rdls\.org\z
```
[the remark this answers is not in the document]
Comment (Dantemss) [reply config/secrets.yml.example]: I think so and ok

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -85,6 +85,13 @@ def create
       success: -> do
         authentication = @handler_result.outputs[:authentication]
         status = @handler_result.outputs[:status]
+
+        # Set the SSO cookie
+        session_config = Rails.application.secrets.sso['cookie']
```
[the remark this answers is not in the document]
Comment (Dantemss) [reply app/controllers/sessions_controller.rb]: OK

**Comment by Dantemss** [comment]
One last thing I want to check: I think I can take out the code that derives a key from the sso secret token and instead use it directly to encrypt the cookies. Might make the code to decypt the cookies simpler.

File: lib/host.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  TRUSTED_HOST_REGEXES = Rails.application.secrets.trusted_hosts.map do |host|
+    /\A(.*\.)?#{host.gsub('.', '\.')}\z/
+  end
+
+  def self.trusted?(url)
+    uri = Addressable::URI.parse url
+
+    TRUSTED_HOST_REGEXES.any? { |regex| regex.match uri.host }
+  end
+end
```
Comment (jpslav) [inline lib/host.rb]: Nice cleanup of this code Dante.

**Comment by Dantemss** [comment]
@jpslav changes made

File: config/initializers/sso_cookies.rb [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/middleware/cookies.rb
 
+# Note: when we upgrade to Rails 5, everyone will be logged out of SSO.
```
Comment (Dantemss) [inline config/initializers/sso_cookies.rb]: If you are explicitly setting the cipher and you are ok with continuing to be on CBC forever, they won't be logged out.

**Pull request merged**